governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -0,0 +1,410 @@
+  }
+
+  public async getStakeAccountCustody(
+    stakeAccountPositions: PublicKey
+  ): Promise<Account> {
+    return getAccount(
+      this.connection,
+      getStakeAccountCustodyAddress(stakeAccountPositions)
+    );
+  }
+
+  public async initializePool({
+    rewardProgramAuthority,
+    poolData,
+    y,
+  }: {
+    rewardProgramAuthority: PublicKey;
+    poolData: PublicKey;
+    y: bigint;
+  }): Promise<void> {
+    const yAnchor = convertBigIntToBN(y);
+    const instruction = await this.integrityPoolProgram.methods
+      .initializePool(rewardProgramAuthority, yAnchor)
+      .accounts({
+        poolData,
+      })
+      .instruction();
+
+    const transactions =
+      await TransactionBuilder.batchIntoVersionedTransactions(
+        this.wallet.publicKey,
+        this.provider.connection,
+        [
+          {
+            instruction,
+            signers: [],
+          },
+        ],
+        {}
+      );
+
+    await sendTransactions(transactions, this.connection, this.wallet);
+  }
+
+  public async getOwnerPythAtaAccount(): Promise<Account> {
+    const globalConfig = await this.getGlobalConfig();
+    return getAccount(
+      this.connection,
+      await getAssociatedTokenAddress(
+        globalConfig.pythTokenMint,
+        this.wallet.publicKey
+      )
+    );
+  }
+
+  public async getPoolConfigAccount(): Promise<PoolConfig> {
+    const poolConfigAnchor =
+      await this.integrityPoolProgram.account.poolConfig.fetch(
+        getPoolConfigAddress()
+      );
+    return convertBNToBigInt(poolConfigAnchor);
+  }
+
+  public async getPoolDataAccount(): Promise<PoolDataAccount> {
+    const poolConfig = await this.getPoolConfigAccount();
+    const poolDataAddress = poolConfig.poolData;
+    const poolDataAccountAnchor =
+      await this.integrityPoolProgram.account.poolData.fetch(poolDataAddress);
+    return convertBNToBigInt(poolDataAccountAnchor);
+  }
+
+  public async getPublishers(): Promise<
+    {
+      pubkey: PublicKey;
+      stakeAccount: PublicKey | null;
+    }[]
+  > {
+    const poolData = await this.getPoolDataAccount();
+
+    return poolData.publishers
+      .map((publisher, index) => ({
+        pubkey: publisher,
+        stakeAccount: poolData.publisherStakeAccounts[index]?.equals(
+          PublicKey.default
+        )
+          ? null
+          : poolData.publisherStakeAccounts[index]!,
+      }))
+      .filter(({ pubkey }) => pubkey && !pubkey.equals(PublicKey.default));
+  }
+
+  public async stakeToGovernance(
+    stakeAccountPositions: PublicKey,
+    amount: bigint
+  ) {
+    const instruction = await this.stakingProgram.methods
+      .createPosition(
+        {
+          voting: {},
+        },
+        new BN(amount.toString())
+      )
+      .accounts({
+        stakeAccountPositions,
+      })
+      .instruction();
+
+    const transactions =
+      await TransactionBuilder.batchIntoVersionedTransactions(
```

Comment:
you could refator this into a function that takes in some instruction and does this step + `sendTransactions`

Reply:
done.